scripts/generate_profile_shell_scripts.py

```diff
@@ -0,0 +1,247 @@
+
+    # Beware, support is poor on the FAIR cluster and Colab machines due to
+    # older Nvidia drivers. For best OpenGL profiling, profile your desktop
+    # linux machine using the Nsight Systems GUI, not the nsys command-line
+    # tool.
+    capture_opengl = False
+
+    # nsys produces a .qdrep multithreaded trace file which can be viewed in the
+    # Nsight GUI. Optionally, it can also export a .sqlite database file for use
+    # with habitat-sim's compare_profiles.py helper script.
+    export_sqlite = True
+
+    # This is the end of the user-facing options, except see slurm_submit_str
+    # below.
+    # =========================================================================
+
+    if do_capture_step_range:
+        program_with_extra_args_str = (
+            program_str
+            + " PROFILING.CAPTURE_START_STEP "
+            + str(capture_start_step)
+            + " PROFILING.NUM_STEPS_TO_CAPTURE "
+            + str(num_steps_to_capture)
+        )
+    else:
+        program_with_extra_args_str = program_str
+
+    if do_capture_step_range:
+        capture_range_args = '--capture-range=nvtx -p "habitat_capture_range" --stop-on-range-end=true'
+    else:
+        capture_range_args = (
+            "--delay="
+            + str(nsys_capture_delay_seconds)
+            + " --duration="
+            + str(nsys_capture_duration_seconds)
+        )
+
+    task_capture_str = (
+        """export HABITAT_PROFILING=1
+export NSYS_NVTX_PROFILER_REGISTER_ONLY=0
+"""
+        + nsys_path
+        + " profile --sample=none --trace-fork-before-exec=true --force-overwrite=true --trace=nvtx"
+        + (",cuda" if capture_cuda else "")
+        + (",opengl" if capture_opengl else "")
+        + " "
+        + capture_range_args
+        + ' --output="'
+        + profile_output_folder
+        + "/"
+        + profile_output_filename_base
+        + '" '
+        + ("--export=sqlite" if export_sqlite else "")
+        + " "
+        + program_with_extra_args_str
+    )
+
+    if do_slurm:
+        if capture_all_tasks:
+            slurm_task_str = (
+                """#!/bin/sh
+"""
+                + task_capture_str
+                + """
+"""
+            )
+        else:
+            slurm_task_str = (
+                """#!/bin/sh
+if [ ${SLURM_NODEID} == "0" ] && [ ${SLURM_LOCALID} == "0" ]
+then
+"""
+                + task_capture_str
+                + """
+else
+"""
+                + program_str
+                + """
+fi"""
+            )
+
+        slurm_submit_str = (
+            """#!/bin/bash
+#SBATCH --job-name=capture_profile
+#SBATCH --output=/checkpoint/%u/jobs/job.%j.out
+#SBATCH --error=/checkpoint/%u/jobs/job.%j.err
+#SBATCH --gpus-per-task 1
+#SBATCH --nodes 8
+#SBATCH --cpus-per-task 10
+#SBATCH --ntasks-per-node 8
+#SBATCH --mem-per-cpu=5GB
+#SBATCH --partition=learnfair
+#SBATCH --time="""
+            + str(slurm_job_termination_minutes)
+            + """:00
+#SBATCH --signal=USR1@300
```

Comment:
This should be removed.  Seems like you want `slurm_job_termination_minutes` to be respected and the signal will make the job exit earlier and requeue itself.  Since the `capture_profile_slurm_task.sh` script doesn't have a signal forwarder this signal won't get propagated, that is why it doesn't matter currently.